selfdrive/controls/lib/pid.py

```diff
@@ -29,10 +31,14 @@ def __init__(self, k_p, k_i, k_f=1., pos_limit=None, neg_limit=None, rate=100, s
 
   @property
   def k_p(self):
+    if isinstance(self._k_p, Number):
+      return self._k_p
```

Comment:
we shouldn't handle two types here. why not just pass in a list?

Reply:
Just a list doesn't work, needs to be a list of lists containing x,y pairs for interpolation based on speed (which doesn't make much sense outside of a vehicle controls context). Had it this way first, but looks pretty ugly